alicloud/resource_alicloud_ess_scalingconfiguration.go

```diff
@@ -286,11 +288,22 @@ func modifyEssScalingConfiguration(d *schema.ResourceData, meta interface{}) err
 		d.SetPartial("image_id")
 	}
 
-	if d.HasChange("instance_type") {
-		instanceTypes := make([]string, 0, 1)
-		instanceTypes = append(instanceTypes, d.Get("instance_type").(string))
-		args.InstanceTypes = &instanceTypes
-		d.SetPartial("instance_type")
+	hasChangeInstanceType := d.HasChange("instance_type")
+	hasChangeInstanceTypes := d.HasChange("instance_types")
+	if hasChangeInstanceType || hasChangeInstanceTypes {
+		instanceType := d.Get("instance_type").(string)
+		instanceTypes := d.Get("instance_types").([]interface{})
+		if instanceType == "" && (instanceTypes == nil || len(instanceTypes) == 0) {
+			return fmt.Errorf("securityGroupId or securityGroupIds must be assigned")
```

Comment:
this should be instance_type or instance_types must be assigned

Reply:
fixed